test/controllers/challenges_controller_test.rb

```diff
@@ -139,6 +165,32 @@ def setup
     assert_equal I18n.t('flag.accepted'), flash[:notice]
   end
 
+  test 'solved challenge survey shows for standard challenge when submitted flag is accepted' do
```

Comment:
These tests feel like integration tests, not controller tests. They should be moved to the integration folder.